### PR TITLE
Transform ranges and simple expressions

### PR DESCRIPTION
**Pull request by yurydelendik**
Depends on https://github.com/CraneStation/cranelift/pull/698

Introduces writing of the .debug_rnglist and .debug_loc sections. The wasm ranges are converted to native ranges, and if needed, further spitted by for expression/var location handling.

TODO:

- [x] refactor inner loops into iterators
- [x] don't assume vmctx stay the same in a var range

### CODE REVIEW

**Comment by yurydelendik** [comment]
Teaser screenshot of debugging wasm in vscode, which run wasmtime and handled by lldb-mi:

Screen Shot 2019-03-15 at 7 05 26 PM

**Comment by yurydelendik** [comment]
depends on #186

**Comment by sunfishcode** [comment]
Thanks!

File: wasmtime-debug/src/transform/expression.rs [hunk 1]
```diff
@@ -0,0 +1,486 @@
...
+    let buf = expr.0.to_slice()?;
+    while !pc.is_empty() {
+        let next = buf[pc.offset_from(&expr.0).into_u64() as usize];
+        need_deref = true;
+        if next == 0xED {
+            // WebAssembly DWARF extension
+            pc.read_u8()?;
+            let ty = pc.read_uleb128()?;
+            assert_eq!(ty, 0);
+            let index = pc.read_sleb128()?;
```
Comment (ggreif) [inline wasmtime-debug/src/transform/expression.rs:314]: I guess this should be `uleb128`

**Pull request merged**